t/unit/backends/test_base.py

```diff
@@ -456,6 +463,17 @@ def test_exception_to_python_when_attribute_exception(self):
         result_exc = b.exception_to_python(test_exception)
         assert str(result_exc) == 'Raise Custom Message'
 
+    def test_exception_to_python_when_type_error(self):
+        b = BaseBackend(app=self.app)
+        celery.TestParamException = paramexception
```

Comment:
You could possibly use [patch.object](https://docs.python.org/3/library/unittest.mock.html#patch-object) as a context manager:

```python
with patch.object(celery, 'TestParamException', return_value=paramexception):
```